device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -273,6 +251,27 @@ public void open() throws TransportException
 
             log.debug("Opening MQTT connection...");
 
+            if (this.config.getSasTokenAuthentication() != null)
+            {
+                try
+                {
+                    log.trace("Setting password for MQTT connection since it is a SAS token authenticated connection");
```

Comment:
Is `open()` called internally each time after a disconnect due to SAS expiration?

Reply:
Yes